### PR TITLE
refactor: convert resize mixin to a modifier

### PR DESCRIPTION
**Pull request by tsmalls93**
https://kubprojects.atlassian.net/browse/FWS-1393

Correct me if I am wrong, but I am guessing that we will wait until the dummy app is converted to add more documentation. I can make a card to do so for this if that is correct.

### CODE REVIEW

File: addon/templates/components/nrg-modal-container/modal-wrapper.hbs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{! template-lint-disable no-inline-styles }}
-<div class="ui modal visible active {{_modalClass}}" style={{modalStyles}} role="dialog" {{did-insert (action this.modalElementAdded)}}>
+<div class="ui modal visible active {{_modalClass}}" style={{modalStyles}} role="dialog" {{did-insert (action this.modalElementAdded)}} {{on-resize this.onResize}}>
```
[the remark this answers is not in the document]
Comment (tsmalls93) [reply addon/templates/components/nrg-modal-container/modal-wrapper.hbs:2]: I will take care of this when I convert this to a glimmer component.

File: addon/modifiers/on-resize.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    this.resizeObserver.observe(this.element, this.handleResize);
+  }
+
+  willRemove() {
+    this.resizeObserver.unobserve(this.element, this.handleResize);
+  }
+
+  @action
+  handleResize(...args) {
+    this.callback(...args);
```
Comment (KeithClinard) [inline addon/modifiers/on-resize.js:22]: The `arguments` property in javascript is implicit even if no arguments are named in the function.
```suggestion
  handleResize() {
    this.callback(...arguments);
```

File: addon/components/nrg-application.hbs [hunk 3]
```diff
@@ -3,8 +3,7 @@
   class="nrg-application
     {{if @fullscreenMap "fullscreen-map"}}
     {{if this.computerScreenSidebarActive "large-screen-sidebar-active"}}"
-  {{did-insert this.onInsert}}
-  {{will-destroy this.onDestroy}}
+  {{on-resize this.setMainContentStyle}}
```
Comment (KeithClinard) [inline addon/components/nrg-application.hbs:6]: `setMainContentStyle` needs the `@action` decorator to preserve the context of `this`

File: addon/components/nrg-responsive-takeover.hbs [hunk 4]
```diff
@@ -4,8 +4,7 @@
   @isOpen={{true}}
   @dismissable={{false}}
   @takeover={{true}}
-  {{did-insert this.onInsert}}
-  {{will-destroy this.onDestroy}}
+  {{on-resize this.setMainContentStyle}}
```
Comment (KeithClinard) [inline addon/components/nrg-responsive-takeover.hbs:7]: `setMainContentStyle` needs the `@action` decorator to preserve the context of `this`

File: addon/components/nrg-sidebar.hbs [hunk 5]
```diff
@@ -1,8 +1,7 @@
 <NrgModal
   @sidebar={{true}}
   @isOpen={{this.application.sidebarIsOpen}}
-  {{did-insert this.onInsert}}
-  {{will-destroy this.onDestroy}}
+  {{on-resize onResize}}
```
Comment (KeithClinard) [inline addon/components/nrg-sidebar.hbs:4]: `onResize` needs the `@action` decorator to preserve the context of `this`

Also:
```suggestion
  {{on-resize this.onResize}}
```

**Review by KeithClinard — CHANGES_REQUESTED** [review]
I love this.
It feels very clean.

**Review by KeithClinard — APPROVED** [review]

**Pull request merged**